src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -12,28 +12,127 @@ Namespace Microsoft.CodeAnalysis.Operations
             Return If(value Is Nothing OrElse value.IsBad, New [Optional](Of Object)(), New [Optional](Of Object)(value.Value))
         End Function
 
-        Private Shared Function GetAssignmentKind(value As BoundAssignmentOperator) As OperationKind
+        Private Shared Function GetAssignmentInfo(value As BoundAssignmentOperator) As (OperationKind As OperationKind, OperatorKind As BinaryOperatorKind, IsChecked As Boolean, InConversion As Conversion, OutConversion As Conversion, RightNode As BoundExpression, OperatorMethod As IMethodSymbol)
+            Dim identityConversion = New Conversion(Conversions.Identity)
```

Comment:
>Dim identityConversion = New Conversion(Conversions.Identity) [](start = 12, length = 61)

It feels like implementation here is more complicated than it could be. 
We need to have two helpers:
1. To dig through a conversion (standard or user-defined)
2. To dig through a binary operator (built-in or user-defined)

Then we can use them to decompose the right node. 
If the top is a conversion, this is an 'out' conversion, get its properties and dig to the operand, which must be a binary operator
Get information about the binary operator (Kind, method, etc ) and dig to the operands
If left operand is a conversion, that is the 'in' conversion, dig through to its operand.
That must be the placeholder.
Done.

It looks like we need to make some small adjustments to Binder.BindCompoundAssignment to enforce the shape of the tree in all scenarios. 
Let's talk offline in more details.